models/things.py

```diff
@@ -184,6 +184,46 @@ def display_name(cls):
         return 'Shelly Humidity'
 
 
+class ShellyTRV(Thing):
+    __mapper_args__ = {
+        'polymorphic_identity': 'shellytrv'
+    }
+
+    def get_data_type(self):
+        return DataType.Float
+
+    def get_state_topic(self):
+        return "shellies/{device_id}/info".format(type=self.type, device_id=self.device_id, vnode_id=self.vnode_id)
+
+    def get_base_topic(self):
+        return "shellies/{device_id}/thermostat/{vnode_id}/command/".format(type=self.type,
+                                                                                    device_id=self.device_id,
+                                                                                    vnode_id=self.vnode_id)
+
+    def get_action_topic(self):
+        return "f{self.get_base_topic()}/target_t"
+
+    def send_external_temperature(self, value):
+        mq.publish(f"{self.get_base_topic()}/ext_t", value)
+
+    def send_value(self, value):
+        # Todo: clamp values if needed
+        mq.publish(self.get_action_topic(), value)
+
+    def process_status(self, db, state):
+        LastSeen.update_last_seen(db, self.device_id)
+        try:
+            data = json.loads(state)
+        except json.JSONDecodeError:
+            ...
+        state = data.get("thermostats")[self.vnode_id].get("target_t").get("value")
```

Comment:
If `json.loads` fails `data` will be `None`. Either initialize before the `try` block or in the `except` block.
Also the index operation based on the `vnode_id` could fail, throwing an `IndexError`.